features/release.feature

```diff
@@ -294,3 +294,42 @@ MARKDOWN
       """
       Attaching release asset `hello-1.2.0.tar.gz'...\n
       """
+
+    Scenario: Download a release asset.
+      Given the GitHub API server:
+        """
+        get('/repos/mislav/will_paginate/releases') {
+          json [
+            { url: 'https://api.github.com/repos/mislav/will_paginate/releases/123',
+              upload_url: 'https://api.github.com/uploads/assets{?name,label}',
+              tag_name: 'v1.2.0',
+              name: 'will_paginate 1.2.0',
+              draft: true,
+              prerelease: false,
+              assets: [
+                { url: 'https://api.github.com/repos/mislav/will_paginate/assets/9876',
+                  name: 'hello-1.2.0.tar.gz',
+                },
+              ],
+            },
+          ]
+        }
+        get('/repos/mislav/will_paginate/assets/9876') {
+          status 302
+          headers['Location'] = 'https://github-cloud.s3.amazonaws.com/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz'
+          ""
+        }
+        get('/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz') {
```

Comment:
Hrmm, maybe the Go client isn't redirecting correctly. I'll try changing the code to match the redirect tooling I see in `Get()` though it might be a few days before I do that.

tcpflow output:
```
127.000.000.001.53925-127.000.000.001.53924: GET /__identify__ HTTP/1.1
Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
Accept: */*
User-Agent: Ruby
Host: 127.0.0.1:53924


127.000.000.001.53925-127.000.000.001.53924: GET /__identify__ HTTP/1.1
Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
Accept: */*
User-Agent: Ruby
Host: 127.0.0.1:53924


127.000.000.001.53924-127.000.000.001.53925: HTTP/1.1 200 OK
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Content-Length: 14
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53925: HTTP/1.1 200 OK
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Content-Length: 14
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53925: 70296040667240
127.000.000.001.53924-127.000.000.001.53925: 70296040667240
127.000.000.001.53928-127.000.000.001.53924: GET /repos/mislav/will_paginate/releases HTTP/1.1
Host: api.github.com
User-Agent: Go-http-client/1.1
Authorization: token OTOKEN
X-Original-Port: 80
X-Original-Scheme: https
Accept-Encoding: gzip


127.000.000.001.53928-127.000.000.001.53924: GET /repos/mislav/will_paginate/releases HTTP/1.1
Host: api.github.com
User-Agent: Go-http-client/1.1
Authorization: token OTOKEN
X-Original-Port: 80
X-Original-Scheme: https
Accept-Encoding: gzip


127.000.000.001.53924-127.000.000.001.53928: HTTP/1.1 200 OK
Content-Type: application/json;charset=utf-8
Content-Length: 332
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53928: HTTP/1.1 200 OK
Content-Type: application/json;charset=utf-8
Content-Length: 332
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53928: [{"url":"https://api.github.com/repos/mislav/will_paginate/releases/123","upload_url":"https://api.github.com/uploads/assets{?name,label}","tag_name":"v1.2.0","name":"will_paginate 1.2.0","draft":true,"prerelease":false,"assets":[{"url":"https://api.github.com/repos/mislav/will_paginate/assets/9876","name":"hello-1.2.0.tar.gz"}]}]
127.000.000.001.53924-127.000.000.001.53928: [{"url":"https://api.github.com/repos/mislav/will_paginate/releases/123","upload_url":"https://api.github.com/uploads/assets{?name,label}","tag_name":"v1.2.0","name":"will_paginate 1.2.0","draft":true,"prerelease":false,"assets":[{"url":"https://api.github.com/repos/mislav/will_paginate/assets/9876","name":"hello-1.2.0.tar.gz"}]}]
127.000.000.001.53929-127.000.000.001.53924: GET /repos/mislav/will_paginate/assets/9876 HTTP/1.1
Host: api.github.com
User-Agent: Go-http-client/1.1
Authorization: token OTOKEN
Content-Type: application/octet-stream
X-Original-Port: 80
X-Original-Scheme: https
Accept-Encoding: gzip


127.000.000.001.53929-127.000.000.001.53924: GET /repos/mislav/will_paginate/assets/9876 HTTP/1.1
Host: api.github.com
User-Agent: Go-http-client/1.1
Authorization: token OTOKEN
Content-Type: application/octet-stream
X-Original-Port: 80
X-Original-Scheme: https
Accept-Encoding: gzip


127.000.000.001.53924-127.000.000.001.53929: HTTP/1.1 302 Found
Content-Type: application/json;charset=utf-8
Location: https://github-cloud.s3.amazonaws.com/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz
Content-Length: 2
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53929: HTTP/1.1 302 Found
Content-Type: application/json;charset=utf-8
Location: https://github-cloud.s3.amazonaws.com/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz
Content-Length: 2
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53929: {}
127.000.000.001.53924-127.000.000.001.53929: {}
127.000.000.001.53929-127.000.000.001.53924: GET /releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz HTTP/1.1
Host: 127.0.0.1:53924
User-Agent: Go-http-client/1.1
Referer: https://api.github.com/repos/mislav/will_paginate/assets/9876
X-Original-Port: 80
X-Original-Scheme: https
Accept-Encoding: gzip


127.000.000.001.53929-127.000.000.001.53924: GET /releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz HTTP/1.1
Host: 127.0.0.1:53924
User-Agent: Go-http-client/1.1
Referer: https://api.github.com/repos/mislav/will_paginate/assets/9876
X-Original-Port: 80
X-Original-Scheme: https
Accept-Encoding: gzip


127.000.000.001.53924-127.000.000.001.53929: HTTP/1.1 200 OK
Content-Type: application/octet-stream
Content-Length: 1238
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53929: HTTP/1.1 200 OK
Content-Type: application/octet-stream
Content-Length: 1238
Server: WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)
Date: Wed, 17 Feb 2016 15:13:53 GMT
Connection: Keep-Alive


127.000.000.001.53924-127.000.000.001.53929: {
  "GATEWAY_INTERFACE": "CGI/1.1",
  "PATH_INFO": "/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz",
  "QUERY_STRING": "",
  "REMOTE_ADDR": "127.0.0.1",
  "REMOTE_HOST": "localhost",
  "REQUEST_METHOD": "GET",
  "REQUEST_URI": "http://127.0.0.1:53924/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz",
  "SCRIPT_NAME": "",
  "SERVER_NAME": "127.0.0.1",
  "SERVER_PORT": "53924",
  "SERVER_PROTOCOL": "HTTP/1.1",
  "SERVER_SOFTWARE": "WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)",
  "HTTP_HOST": "127.0.0.1:53924",
  "HTTP_USER_AGENT": "Go-http-client/1.1",
  "HTTP_REFERER": "https://api.github.com/repos/mislav/will_paginate/assets/9876",
  "HTTP_X_ORIGINAL_PORT": "80",
  "HTTP_X_ORIGINAL_SCHEME": "https",
  "HTTP_ACCEPT_ENCODING": "gzip",
  "rack.version": [
    1,
    1
  ],
  "rack.input": "#<StringIO:0x007fde2505e988>",
  "rack.errors": "#<IO:0x007fde240c6520>",
  "rack.multithread": true,
  "rack.multiprocess": false,
  "rack.run_once": false,
  "rack.url_scheme": "http",
  "HTTP_VERSION": "HTTP/1.1",
  "REQUEST_PATH": "/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz",
  "rack.logger": "#<Rack::NullLogger:0x007fde2510f120>",
  "rack.request.query_string": "",
  "rack.request.query_hash": {
  }
}
127.000.000.001.53924-127.000.000.001.53929: {
  "GATEWAY_INTERFACE": "CGI/1.1",
  "PATH_INFO": "/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz",
  "QUERY_STRING": "",
  "REMOTE_ADDR": "127.0.0.1",
  "REMOTE_HOST": "localhost",
  "REQUEST_METHOD": "GET",
  "REQUEST_URI": "http://127.0.0.1:53924/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz",
  "SCRIPT_NAME": "",
  "SERVER_NAME": "127.0.0.1",
  "SERVER_PORT": "53924",
  "SERVER_PROTOCOL": "HTTP/1.1",
  "SERVER_SOFTWARE": "WEBrick/1.3.1 (Ruby/2.2.3/2015-08-18)",
  "HTTP_HOST": "127.0.0.1:53924",
  "HTTP_USER_AGENT": "Go-http-client/1.1",
  "HTTP_REFERER": "https://api.github.com/repos/mislav/will_paginate/assets/9876",
  "HTTP_X_ORIGINAL_PORT": "80",
  "HTTP_X_ORIGINAL_SCHEME": "https",
  "HTTP_ACCEPT_ENCODING": "gzip",
  "rack.version": [
    1,
    1
  ],
  "rack.input": "#<StringIO:0x007fde2505e988>",
  "rack.errors": "#<IO:0x007fde240c6520>",
  "rack.multithread": true,
  "rack.multiprocess": false,
  "rack.run_once": false,
  "rack.url_scheme": "http",
  "HTTP_VERSION": "HTTP/1.1",
  "REQUEST_PATH": "/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz",
  "rack.logger": "#<Rack::NullLogger:0x007fde2510f120>",
  "rack.request.query_string": "",
  "rack.request.query_hash": {
  }
}
```